src/main/java/reposense/util/TimeUtil.java

```diff
@@ -66,4 +66,13 @@ public static int getZoneRawOffset(ZoneId zoneId) {
         ZoneOffset zoneOffset = zoneId.getRules().getOffset(now);
         return zoneOffset.getTotalSeconds() * 1000;
     }
+
+    /**
+     * Get the {@code current} date that is in {@code fromZoneId} timezone into {@code toZoneId} timezone.
+     */
+    public static Date convertDateWithZones(Date current, ZoneId fromZoneId, ZoneId toZoneId) {
```

Comment:
We might be able to use the functionality in this function for `getSystemDateFromZonedDate`, since it is using the same logic.

Reply:
I think we should stick to using `getSystemDateFromZonedDate`? Right now I think the `fromZoneId` will always be the system default, which was the original implementation.